ygot/struct_validation_map_test.go

```diff
@@ -1397,10 +1398,28 @@ func TestMergeStructs(t *testing.T) {
 		inA:     &validatedMergeTest{String: String("karbach-hopadillo")},
 		inB:     &validatedMergeTest{String: String("blackwater-draw-brewing-co-border-town")},
 		wantErr: "error merging b to new struct: destination value was set when merging, src: blackwater-draw-brewing-co-border-town, dst: karbach-hopadillo",
+	}, {
+		name:   "allow leaf overwrite if equal",
+		inA:    &validatedMergeTest{String: String("new-belgium-sour-saison")},
+		inB:    &validatedMergeTest{String: String("new-belgium-sour-saison")},
+		inOpts: []MergeOption{NewAllowLeafOverwrite(true)},
+		want:   &validatedMergeTest{String: String("new-belgium-sour-saison")},
+	}, {
+		name:   "allow leaf overwrite if unequal",
+		inA:    &validatedMergeTest{String: String("sierra-nevada-sidecar")},
+		inB:    &validatedMergeTest{String: String("deschutes-pacific-wonderland")},
+		inOpts: []MergeOption{NewAllowLeafOverwrite(false)},
+		want:   &validatedMergeTest{String: String("deschutes-pacific-wonderland")},
+	}, {
+		name:    "error - merge leaf overwrite but not equal",
+		inA:     &validatedMergeTest{String: String("schneider-weisse-hopfenweisse")},
+		inB:     &validatedMergeTest{String: String("deschutes-jubelale")},
+		inOpts:  []MergeOption{NewAllowLeafOverwrite(true)},
+		wantErr: "error merging b to new struct: destination value was not equal to source when merging with override, src: deschutes-jubelale, dst: schneider-weisse-hopfenweisse",
 	}}
```

Comment:
Missing case of no override but values are equal. I think this should pass but if not I'd suggest putting a comment why not since that is not intuitive.

Reply:
Today it won't -- because the default behaviour is to never allow override.